### PR TITLE
docs: use of datasets for airflow >= 2.4

### PR DESCRIPTION
**Pull request by SiddiqueAhmad**
schedule is right parameter name

## Description

schedule is right parameter name. 

## Checklist

- [ ] I have made corresponding changes to the documentation (if required)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by tatiana — COMMENTED** [review]
Hi @SiddiqueAhmad thanks for your contribution!

Until Airflow 2.3, `schedule_interval` didn't exist - only `schedule`. Since Cosmos supports Airflow 2.3, we still stick to `schedule` in our example DAGs.

Please, rename the PR - since this is not a correction per say.

That said, I'm happy for us to update the docs to illustrate the usage in the more recent versions of Airflow. As part of this, could you confirm if there are any other parts of the docs that used `schedule_interval`? As long as we don't change the dags in `dev/dags`, we should be good.

I left a minor comment inline

File: docs/configuration/scheduling.rst [hunk 1]
```diff
@@ -50,7 +50,7 @@ Then, you can use Airflow's data-aware scheduling capabilities to schedule ``my_
 
     project_two = DbtDag(
         # ...
-        schedule_interval=[get_dbt_dataset("my_conn", "project_one", "my_model")],
+        schedule=[get_dbt_dataset("my_conn", "project_one", "my_model")],
```
[the remark this answers is not in the document]
Comment (SiddiqueAhmad) [reply docs/configuration/scheduling.rst:55]: Line 48 shows a time based scheduling where `schedule_interval` is right parameter name. What do you say?

File: docs/configuration/scheduling.rst [hunk 2]
```diff
@@ -49,8 +49,10 @@ Then, you can use Airflow's data-aware scheduling capabilities to schedule ``my_
     )
 
     project_two = DbtDag(
-        # ...
-        schedule_interval=[get_dbt_dataset("my_conn", "project_one", "my_model")],
+        # for airflow >=2.3
```
Comment (tatiana) [inline docs/configuration/scheduling.rst:52]:
```suggestion
        # for airflow <=2.3
```

File: docs/configuration/scheduling.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (tatiana) [reply docs/configuration/scheduling.rst:55]: @SiddiqueAhmad The param `schedule_interval` was deprecated and will be removed in a future release of Airflow, they recommend using `schedule` instead:
https://github.com/apache/airflow/blob/a0c19850891bd11bbcdac95742efb3ea4f93c26e/airflow/models/dag.py#L574-L578

File: docs/configuration/scheduling.rst (hunk 2 above)
Comment (SiddiqueAhmad) [reply docs/configuration/scheduling.rst]: Ops, fixed

File: docs/configuration/scheduling.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (SiddiqueAhmad) [reply docs/configuration/scheduling.rst:55]: changed as per suggestion

**Review by jlaneve — APPROVED** [review]
great catch, thank you @SiddiqueAhmad!

**Pull request merged**